client/app/scripts/charts/nodes-layout.js

```diff
@@ -481,7 +481,8 @@ export function doLayout(immNodes, immEdges, opts) {
 
   if (layout) {
     // Last line of defense - re-render everything if two nodes are too close to one another.
-    if (minEuclideanDistanceBetweenPoints(layout.nodes) < NODE_CENTERS_SEPARATION_FACTOR) {
+    if (layout.nodes.count() > 1 &&
```

Comment:
I just discovered that Javascript has `Infinity`! :) Maybe it would be easier to simplify `minEuclideanDistanceBetweenPoints` by setting `let minDistance = Infinity;` there (and then removing `foundPair` and extra logic that it introduces). That would also make this check unnecessary.